### PR TITLE
Upgrade Rails 5.2.0 => 6.0.2

### PR DESCRIPTION
**Pull request by edwardkerry**
Upgrades Rails from 5.2.0 to 6.2.0

See individual commit messages for more details

[trello](https://trello.com/c/yzPWe3yp/1819-5-upgrade-collections-publisher-to-rails-6)

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -1,29 +1,36 @@
...
+# require "active_storage/engine"
 require "action_controller/railtie"
+# require "action_mailer/railtie"
+# require "action_mailbox/engine"
+# require "action_text/engine"
 require "action_view/railtie"
+# require "action_cable/engine"
 require "sprockets/railtie"
-
-require "tilt/erubi"
```
Comment (AlanGabbianelli) [inline config/application.rb]: Why don't we need this anymore?

File: config/application.rb [hunk 2]
```diff
@@ -1,29 +1,36 @@
 require_relative "boot"
 
-require "active_job/railtie"
+require "rails"
```
Comment (AlanGabbianelli) [inline config/application.rb]:
```suggestion
```
We don't need to require rails. We are already requiring the specific frameworks we need.

File: config/application.rb [hunk 3]
```diff
@@ -1,29 +1,36 @@
...
 require "action_controller/railtie"
+# require "action_mailer/railtie"
+# require "action_mailbox/engine"
+# require "action_text/engine"
 require "action_view/railtie"
+# require "action_cable/engine"
 require "sprockets/railtie"
-
-require "tilt/erubi"
+require "rails/test_unit/railtie"
```
Comment (AlanGabbianelli) [inline config/application.rb]: Why do we need this?

File: config/environments/production.rb [hunk 4]
```diff
@@ -14,19 +14,21 @@
...
+  # config.require_master_key = true
+
   # Disable serving static files from the `/public` folder by default since
   # Apache or NGINX already handles this.
   config.public_file_server.enabled = ENV["RAILS_SERVE_STATIC_FILES"].present?
 
-  # Compress JavaScripts and CSS.
+  # Compress CSS using a preprocessor.
   config.assets.js_compressor = :uglifier
   # config.assets.css_compressor = :sass
```
Comment (AlanGabbianelli) [inline config/environments/production.rb]:
```suggestion
  # Compress CSS using a preprocessor.
  # config.assets.css_compressor = :sass
  
  # Compress JavaScript.
  config.assets.js_compressor = :uglifier
```

File: config/initializers/new_framework_defaults_6_0.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+# will be removed in Rails 6.1. This setting is not backwards compatible with earlier Rails versions.
+# If you send mail in the background, job workers need to have a copy of
+# MailDeliveryJob to ensure all delivery jobs are processed properly.
+# Make sure your entire app is migrated and stable on 6.0 before using this setting.
+# Rails.application.config.action_mailer.delivery_job = "ActionMailer::MailDeliveryJob"
+
+# Enable the same cache key to be reused when the object being cached of type
+# `ActiveRecord::Relation` changes by moving the volatile information (max updated at and count)
+# of the relation's cache key into the cache version to support recycling cache key.
+# Rails.application.config.active_record.collection_cache_versioning = true
```
Comment (AlanGabbianelli) [inline config/initializers/new_framework_defaults_6_0.rb]: This file can be safely deleted.

File: config/application.rb [hunk 6]
```diff
@@ -1,29 +1,36 @@
...
 
 # Require the gems listed in Gemfile, including any gems
 # you've limited to :test, :development, or :production.
 Bundler.require(*Rails.groups)
 
 module CollectionsPublisher
   class Application < Rails::Application
```
Comment (AlanGabbianelli) [inline config/application.rb]:
```suggestion
  class Application < Rails::Application
    config.load_defaults 6.0
```
We should add this line to use all the new rails 6 defaults. If not all can be used, we should still add this line and then override below the ones we don't want.

File: config/application.rb (hunk 2 above)
Comment (edwardkerry) [reply config/application.rb]: :+1:

File: config/initializers/new_framework_defaults_6_0.rb (hunk 5 above)
Comment (edwardkerry) [reply config/initializers/new_framework_defaults_6_0.rb]: :+1:

File: config/application.rb (hunk 6 above)
Comment (edwardkerry) [reply config/application.rb:24]: :+1:

File: config/initializers/zeitwerk.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+Rails.autoloaders.each do |autoloader|
+  autoloader.inflector.inflect(
+    "publishing_api_notifier" => "PublishingAPINotifier",
+  )
+end
```
Comment (AlanGabbianelli) [inline config/initializers/zeitwerk.rb:5]: Ah, nice!

File: Gemfile [hunk 8]
```diff
@@ -34,7 +34,7 @@ group :development, :test do
   gem "nokogiri"
   gem "parser", "2.7.1.0"
   gem "pry-byebug"
-  gem "rspec-rails"
+  gem "rspec-rails", "~> 4.0.0.beta4"
```
Comment (AlanGabbianelli) [inline Gemfile:37]: v4 has been released: https://github.com/rspec/rspec-rails/releases/tag/v4.0.0

Comment (edwardkerry) [reply Gemfile]: dropped this commit :+1:

**Pull request merged**